controllers/openstackdataplanerole_controller.go

```diff
@@ -209,6 +209,19 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 
 	r.Log.Info("Role", "DeployStrategy", instance.Spec.DeployStrategy.Deploy, "Role.Namespace", instance.Namespace, "Role.Name", instance.Name)
 	if instance.Spec.DeployStrategy.Deploy {
+
+		// create deployIdentifier
+		if len(instance.Spec.DeployStrategy.DeployIdentifier) == 0 {
+			// patch := client.MergeFrom(instance.DeepCopy())
+			newIdentifier := dataplanev1beta1.GenerateDeployIdentifier()
+			instance.Spec.DeployStrategy.DeployIdentifier = newIdentifier
```

Comment:
Same as above, let's update the Status here rather than the Spec. We can a different path in situations where the user provides a string.

Reply:
I think we would want the value on the Spec to equal the value in the Status to indicate that's the deployment in progress wouldn't we? Otherwise, we would keep hitting these lines since the Spec value would always length of 0.

Reply:
`Spec` shouldn't reflect the value of the `Status` field in my opinion but vice versa.

`Spec.deployIdentifier` should be what we want -> if it's defined in the CR, we use it, otherwise it will be created as a random UUID.

`Status.deployIdentifier` should be valued with the value of `Spec.deployIdentifier` currently running.

My doubts here are:

* If the deploy is completed the `Status.deployIdentifier` field should be resetted to `""` just like `Spec.deployIdentifier`?
* If there's a mismatch between those 2 fields, what should happen? At the moment `.Spec.deployIdentifier` is passed down to AnsibleEE CR as label, and due to https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/pull/145 , if it changes in the AnsibleEE CR, it will force a job recreation (after its/their completion) because the hash is changed. So we could:
  * overwrite the `Status.deployIdentifier` field with the content of `Spec.deployIdentifier` logging to the user that there's an existing deploy that will be overwritten
  * ignore(?) the new CR throwing an error that warns the user that there is a deployment in progress

What do you think folks?

Reply:
I'm not opposed to tracking this in the Status. If it makes the implementation easier to follow, I'm all for it.